index.md: Added text and content to title page/index.md (#38)
ESMValGroup/tutorial: index.md

1. This tutorial can be taken online independently or taught by one of our
   instructors.

2. Don’t be alarmed if you can’t work through the entire tutorial in one
   sitting. It may take some time to get used to working with ESMValTool.

3. If you get stuck, help is always available from the tutors, from ESMValTool
    developers via the [github issues
    page](https://github.com/ESMValGroup/ESMValTool/issues) or via the
    [ESMValTool email list](mailto:esmvaltool@listserv.dfn.de).

4. This tutorial includes several advanced lessons after the conclusion. These
   advanced lessons should be treated like “mini-tutorials”, and include aspects
   like “developing your own diagnostic” or “how to include observations”.

> ## Additional Resources
>
> - [Read the docs page](https://esmvaltool.readthedocs.io/)
> - [ESMValTool home page](https://www.esmvaltool.org/)
> - [Technical description paper](https://doi.org/10.5194/gmd-13-1179-2020)
> - [ESMValTool Source code](https://github.com/ESMValGroup/ESMValTool)
> - [ESMValCore Source code](https://github.com/ESMValGroup/ESMValCore)
>
{: .callout}

## How to cite ESMValTool

Please cite this tutorial as:

> - ESMValTool 2.0 tutorial, ESMValTool developers, version [ADD VERSION],

Author: Is there a way to find to a specific version of the tutorial, or perhaps just some info on where users can get that version info?

Reviewer: At some point we'll have to make a release. Then we can set up zenodo integration, which will provide citation info for a specific release.

Reviewer 2: And i think once we have the zenodo version we also don't need the date anymore, so maybe remove the [DATE] tag?

layout: lesson
root: .  # Is the only page that doesn't follow the pattern /:path/index.html
permalink: index.html  # Is the only page that doesn't follow the pattern /:path/index.html
date: '`r format(Sys.time(), "%d %B, %Y")`'
---

This tutorial is built to teach you how to use ESMValTool.  

> The Earth System Model Evaluation Tool (ESMValTool) is a community developped software toolkit that aims to faciliate

Reviewer 2: ```suggestion
The Earth System Model Evaluation Tool (ESMValTool) is a community developed software toolkit that aims to facilitate
```

the diagnosis and evaluation of the causes and effects of model biases and inter-model spread within the CMIP model
ensemble.

> ## What will you learn in this course
>
>   - What is ESMValTool
>
>   - How to install ESMValTool
>
>   - How to run ESMValTool
>
>   - How to develop your own diagnostics and recipes
>
>   - How to contribute your recipes and diagnostics back into ESMValTool
>
{: .checklist}

> ## Prerequisites
>

> > - Basic understanding of git (optional - but useful)
> >
> >  - Basic understanding of your preferred command line interface (ie a bash terminal)
> >
> >  - Access to CMIP data
> >
> >  - Access to a suitable computing system (eg Jasmin)
> >
> > - Github account (optional, but useful!)

Reviewer 2: Single line breaks do not render in markdown. I would suggest to make a list or two separate lists, with optional and minimal requirements. Does the suitable computing system belong to the optional or minimal list? I would say you own laptop would also work for simple and small recipes and especially this tutorial should be doable on a home computer. Don't we need conda installed for this tutorial?

```suggestion
> *Minimal requirements:*
>   - Basic understanding of your preferred command line interface (i.e. a bash terminal)
>   - Access to CMIP data
>   - Laptop/desktop with [(mini)conda](https://docs.conda.io/en/latest/miniconda.html) installed
> 
> *Optional, but useful:*
>   - Access to a suitable computing system (eg Jasmin)
>   - Basic understanding of git
>   - Github account
```

Reviewer 2: @ledm what do you think about this? My comment keeps disappearing because the text gets updated, but I still have the same questions with this list.

Author: I like this, sorry it keeps gettings pushed back. I'll try to get it in when I can.

Author: Added this in now

{: .prereq}

> ## Main things you need to know before starting this course
>
> 1. This tutorial can be taken online independently or taught by one of our instructors.
> 2. Please check the common issues page if you get stuck [Link to common issues page]. Otherwise, help is always available from ESMValTool developers via the github issues page.  
> 3. This tutorial includes several advanced lessons after the conclusion. These advanced lessons should be treated like “mini-tutorials”, and include aspects like “developing your own diagnostic” or “how to include observations”.
> 4. Don’t be alarmed if you can’t work through the entire tutorial in one sitting. It may take some time to get used to working with ESMValTool.
{: .checklist}

{% include links.md %}

> ## Additional resources: 
> - [Read the docs page](https://esmvaltool.readthedocs.io/)
> - [ESMValTool home page](https://www.esmvaltool.org/)
> - [Technical description paper](https://doi.org/10.5194/gmd-13-1179-2020)
> - [Source code (ESMValTool)](https://github.com/ESMValGroup/ESMValTool)
> - [Source code (ESMValCore )](https://github.com/ESMValGroup/ESMValCore)
{: .callout}

## Citation
Please cite this tutorial as:

> - ESMVAlTool 2.0 tutorial, ESMValtool developpers, version [ADD VERSION], 

Reviewer 2: ```suggestion
- ESMValTool 2.0 tutorial, ESMValTool developers, version [ADD VERSION], 
```

layout: lesson
root: .  # Is the only page that doesn't follow the pattern /:path/index.html
permalink: index.html  # Is the only page that doesn't follow the pattern /:path/index.html
date: '`r format(Sys.time(), "%d %B, %Y")`'
---

This tutorial is built to teach you how to use ESMValTool.  

The Earth System Model Evaluation Tool (ESMValTool) is a community developped software toolkit that aims to faciliate
the diagnosis and evaluation of the causes and effects of model biases and inter-model spread within the CMIP model
ensemble.

This tutorial is structured such that the main body of the tutorial, up to the episode 7, can be done in one sitting.

> From there, there are several mini-tutorials which each covers one aspect of working with ESMValTool.

Reviewer 2: ```suggestion
From episode 8, each episode is a mini-tutorial covering an advanced aspect of working with ESMValTool.
```

These mini-tutorials can be appended to the main tutorial or worked through independently.

> ## What will you learn in this course
>
>   - What is ESMValTool
>
>   - How to install ESMValTool
>   - How to configure ESMValTool for your local system
>   - How to run ESMValTool
>   - How to work with ESMValTool's suite of preprocessors
>   - How to debug your recipes
>   - How to access and deploy recipes from the ESMValTools gallery (Advanced)
>   - How to develop your own diagnostics and recipes (Advanced)
>   - How to contribute your recipes and diagnostics back into ESMValTool (Advanced)
>   - How to include new observational datasets (Advanced)
>

> {: .checklist}

Reviewer 2: I would say this list is already covered by the schedule that is rendered automatically at the bottom.

Author: I think it's good to be explicit and repeat this information.

Reviewer 2: @ledm alright!

> 2. Please check the common issues page if you get stuck [Link to common issues page]. Otherwise, help is always available from ESMValTool developers via the github issues page.  
> 3. This tutorial includes several advanced lessons after the conclusion. These advanced lessons should be treated like “mini-tutorials”, and include aspects like “developing your own diagnostic” or “how to include observations”.
> 4. Don’t be alarmed if you can’t work through the entire tutorial in one sitting. It may take some time to get used to working with ESMValTool.
{: .checklist}

{% include links.md %}

> ## Additional resources: 
> - [Read the docs page](https://esmvaltool.readthedocs.io/)
> - [ESMValTool home page](https://www.esmvaltool.org/)
> - [Technical description paper](https://doi.org/10.5194/gmd-13-1179-2020)
> - [Source code (ESMValTool)](https://github.com/ESMValGroup/ESMValTool)
> - [Source code (ESMValCore )](https://github.com/ESMValGroup/ESMValCore)
{: .callout}

## Citation

Please cite this tutorial as:

- ESMValTool 2.0 tutorial, ESMValTool developers, version [ADD VERSION], 
  [https://github.com/ESMValGroup/tutorial](https://github.com/ESMValGroup/tutorial), 

>   [DATE]. 

Author: It would be nice if this date was displayed as the current date. Anyone have any ideas how to do this?

Similarly, in the line above, it we could include the version number for this tutorial, that would be great too. 

@valeriupredoi any ideas?

Reviewer 2: From what I understand, with the carpentries framework it is only possible to use Rmarkdown in the episodes and not in the title page. This would mean we cannot have dynamically updating variables there.

Author: Okay, perhaps we could have it the about.md citation section?

Reviewer 2: I think we want to have the zenodo version in place of `[ADD VERSION]` and remove the date at the end, because once we have a version, the date is covered.

layout: lesson
root: .  # Is the only page that doesn't follow the pattern /:path/index.html
permalink: index.html  # Is the only page that doesn't follow the pattern /:path/index.html
date: '`r format(Sys.time(), "%d %B, %Y")`'
---

> This tutorial is built to teach you how to use ESMValTool.  

Reviewer 3: ```suggestion
This tutorial helps you to use ESMValTool.  
```

> 2. Please check the common issues page if you get stuck [Link to common issues page]. Otherwise, help is always available from ESMValTool developers via the github issues page.  
> 3. This tutorial includes several advanced lessons after the conclusion. These advanced lessons should be treated like “mini-tutorials”, and include aspects like “developing your own diagnostic” or “how to include observations”.
> 4. Don’t be alarmed if you can’t work through the entire tutorial in one sitting. It may take some time to get used to working with ESMValTool.
{: .checklist}

{% include links.md %}

> ## Additional resources: 
> - [Read the docs page](https://esmvaltool.readthedocs.io/)
> - [ESMValTool home page](https://www.esmvaltool.org/)
> - [Technical description paper](https://doi.org/10.5194/gmd-13-1179-2020)
> - [Source code (ESMValTool)](https://github.com/ESMValGroup/ESMValTool)
> - [Source code (ESMValCore )](https://github.com/ESMValGroup/ESMValCore)
{: .callout}

## Citation

Please cite this tutorial as:

- ESMValTool 2.0 tutorial, ESMValTool developers, version [ADD VERSION], 
  [https://github.com/ESMValGroup/tutorial](https://github.com/ESMValGroup/tutorial), 

>   $date. 

Author: ```suggestion
  [DATE]. 
```

layout: lesson
root: .  # Is the only page that doesn't follow the pattern /:path/index.html
permalink: index.html  # Is the only page that doesn't follow the pattern /:path/index.html

> date: '`r format(Sys.time(), "%d %B, %Y")`'

Reviewer 2: I believe this can go now? We don't use the date variable as the page is not rendered with Rmarkdown.

Reviewer 2: @ledm do you agree?

---

This tutorial helps you to use ESMValTool.  

The Earth System Model Evaluation Tool (ESMValTool) is a community developed software toolkit that aims to facilitate
the diagnosis and evaluation of the causes and effects of model biases and inter-model spread within the CMIP model
ensemble.

This tutorial is structured such that the main body of the tutorial, up to the episode 7, can be done in one sitting.
From episode 8, each episode is a mini-tutorial covering an advanced aspect of working with ESMValTool.
These mini-tutorials can be appended to the main tutorial or worked through independently.

> ## What will you learn in this course
>
>   - What is ESMValTool
>   - How to install ESMValTool
>   - How to configure ESMValTool for your local system
>   - How to run ESMValTool
>   - How to work with ESMValTool's suite of preprocessors
>   - How to debug your recipes
>   - How to access and deploy recipes from the ESMValTools gallery (Advanced)
>   - How to develop your own diagnostics and recipes (Advanced)
>   - How to contribute your recipes and diagnostics back into ESMValTool (Advanced)
>   - How to include new observational datasets (Advanced)
>
{: .checklist}

> > ## Prerequisites

Reviewer 2: There is a codacy issue here about a white space in a block quote. I think the problem is two consecutive quote blocks (what will you learn in this course & prerequisites). And it can be solved by adding a comment:
```suggestion
<!-- -->
> ## Prerequisites
```

root: .  # Is the only page that doesn't follow the pattern /:path/index.html
permalink: index.html  # Is the only page that doesn't follow the pattern /:path/index.html
---

This tutorial is built to teach you ESMValTool.  

> ## What will you learn in this course
>
>   - What is ESMValTool
>
>   - How to install ESMValTool
>
>   - How to run ESMValTool
>
>   - How to develop your own diagnostics and recipes
>
>   - How to contribute your recipes and diagnostics back into ESMValTool
>
{: .checklist}

> ## Prerequisites
>
> Basic understanding of git (optional - but useful)
> Basic understanding of your preferred command line interface (ie a bash terminal)
> Access to CMIP data

> > Access to a suitable computing system (egie jasmin)

Reviewer 4: Should be either "eg" or "ie"?
And should "jasmin" be renamed "Jasmin"?
Do we want to add any other computing system here that people who do not recognize "Jasmin" might understand?

Author: What is the name of DLR's machine?

Reviewer 4: At DLR we run stuff on Mistral at DKRZ.

> Github account (optional, but useful!)
{: .prereq}

> ## Main things you need to know before starting this course
>
> 1. This tutorial can be taken online independently or taught by one of our instructors.

> > 2. Please check the common issues page if you get stuck. Otherwise, help is always available from ESMValTool developers via the github issues page.  

Reviewer 4: Add the link here to the common issues page?

Author: Do we still have a common issues page?

> 3. This tutorial includes several advanced lessons after the conclusion. These advanced lessons should be treated like “mini-tutorials”, and include aspects like “developing your own diagnostic” or “how to include observations”.
> 4. Don’t be alarmed if you can’t work through the entire tutorial in one sitting. It may take some time to get used to working with ESMValTool.
{: .checklist}

{% include links.md %}

> ## Additional resources: 
> 
> - Read the docs page.
> - github main page.
> - GMD paper
> - ESMValTool home page
{: .callout}

> 

Reviewer 4: Do we want to add the links here to the individual resources?

Author: Done!

root: .  # Is the only page that doesn't follow the pattern /:path/index.html
permalink: index.html  # Is the only page that doesn't follow the pattern /:path/index.html
---

> This tutorial is built to teach you ESMValTool.  

Reviewer 2: I would add a little more explanation about the tutorial and ESMValTool. One or two sentences what ESMValTool is so that people that stumble on this page via a shared link or search engine can also find out what it is about. 
I think it is also good to mention that this tutorial is meant to teach the basics of ESMValTool, or if we also add advanced sections, to tell people about this distinction and how to use this tutorial. (e.g. "Section 0 trough 7 are meant to teach the basics and get you started on using ESMValTool, these sections are meant to be completed one after another as a complete tutorial. For more advanced topics like writing your own diagnostic scripts or contributing to the community, we refer you to the advanced sections, which can be done in any order.)

> ## What will you learn in this course
>

> >   - What is ESMValTool

Author: this list hasn't displayed properly

>
>   - How to install ESMValTool
>
>   - How to run ESMValTool
>
>   - How to develop your own diagnostics and recipes
>
>   - How to contribute your recipes and diagnostics back into ESMValTool
>
{: .checklist}

> ## Prerequisites
>
> Basic understanding of git (optional - but useful)
> Basic understanding of your preferred command line interface (ie a bash terminal)
> Access to CMIP data
> Access to a suitable computing system (eg Jasmin)

> > Github account (optional, but useful!)

Reviewer 2: Single line breaks do not render in markdown. I would suggest to make a list or two separate lists, with optional and minimal requirements. Does the suitable computing system belong to the optional or minimal list? I would say you own laptop would also work for simple and small recipes and especially this tutorial should be doable on a home computer. 
```suggestion
> *Minimal requirements:*
>   - Basic understanding of your preferred command line interface (i.e. a bash terminal)
>   - Access to CMIP data
>   - Access to a suitable computing system (eg Jasmin)
>
> *Optional, but useful:*
>   - Basic understanding of git
>   - Github account
```

{: .prereq}

> > ## Main things you need to know before starting this course
> >
> > 1. This tutorial can be taken online independently or taught by one of our instructors.
> > 2. Please check the common issues page if you get stuck [Link to common issues page]. Otherwise, help is always available from ESMValTool developers via the github issues page.  
> > 3. This tutorial includes several advanced lessons after the conclusion. These advanced lessons should be treated like “mini-tutorials”, and include aspects like “developing your own diagnostic” or “how to include observations”.
> > 4. Don’t be alarmed if you can’t work through the entire tutorial in one sitting. It may take some time to get used to working with ESMValTool.
> {: .checklist}

Reviewer 2: I would not put this in a box but just in the main content: 
```suggestion

### Main things you need to know before starting this course

1. This tutorial can be taken online independently or taught by one of our instructors.
2. Please check the common issues page if you get stuck [Link to common issues page]. Otherwise, help is always available from ESMValTool developers via the github issues page.  
3. This tutorial includes several advanced lessons after the conclusion. These advanced lessons should be treated like “mini-tutorials”, and include aspects like “developing your own diagnostic” or “how to include observations”.
4. Don’t be alarmed if you can’t work through the entire tutorial in one sitting. It may take some time to get used to working with ESMValTool.

```

{% include links.md %}

> ## Additional resources: 
> 

> > - Read the docs page: https://esmvaltool.readthedocs.io/

Author: Links don't work here.

> - ESMValTool home page: https://www.esmvaltool.org/
> - Technical description paper: https://doi.org/10.5194/gmd-13-1179-2020 
> - Source code (ESMValTool): https://github.com/ESMValGroup/ESMValTool
> - Source code (ESMValCore ): https://github.com/ESMValGroup/ESMValCore 
{: .callout}

Please cite this tutorial as:

> ESMVAlTool 2.0 tutorial, website, version [ADD AUTO VERSION?] 

Author: This isn't right. Shouldn't we have the GMD paper here?

   instructors.

2. Don’t be alarmed if you can’t work through the entire tutorial in one
   sitting. It may take some time to get used to working with ESMValTool.

3. If you get stuck, help is always available from the tutors, from ESMValTool
    developers via the [github issues
    page](https://github.com/ESMValGroup/ESMValTool/issues) or via the
    [ESMValTool email list](mailto:esmvaltool@listserv.dfn.de).

4. This tutorial includes several advanced lessons after the conclusion. These
   advanced lessons should be treated like “mini-tutorials”, and include aspects
   like “developing your own diagnostic” or “how to include observations”.

> ## Additional Resources
>
> - [Read the docs page](https://esmvaltool.readthedocs.io/)
> - [ESMValTool home page](https://www.esmvaltool.org/)
> - [Technical description paper](https://doi.org/10.5194/gmd-13-1179-2020)
> - [ESMValTool Source code](https://github.com/ESMValGroup/ESMValTool)
> - [ESMValCore Source code](https://github.com/ESMValGroup/ESMValCore)
>
{: .callout}

## How to cite ESMValTool

Please cite this tutorial as:

- ESMValTool 2.0 tutorial, ESMValTool developers, version [ADD VERSION],

>   [https://github.com/ESMValGroup/tutorial](https://github.com/ESMValGroup/tutorial),
>   [DATE].

Reviewer 2: ```suggestion
  [https://github.com/ESMValGroup/tutorial](https://github.com/ESMValGroup/tutorial).
```
